src/SFA.DAS.ApprenticeCommitments.Api/Controllers/ConfirmTrainingProviderController.cs

```diff
@@ -19,13 +19,14 @@ public class ConfirmTrainingProviderController : ControllerBase
         public ConfirmTrainingProviderController(IMediator mediator) => _mediator = mediator;
 
         [HttpPost("apprentices/{apprenticeId}/apprenticeships/{apprenticeshipId}/TrainingProviderConfirmation")]
-        public async Task<IActionResult> ConfirmTrainingProvider(
-            Guid apprenticeId, long apprenticeshipId,
+        [HttpPost("apprentices/{apprenticeId}/apprenticeships/{apprenticeshipId}/{commitmentStatementId}/TrainingProviderConfirmation")]
+        public async Task ConfirmTrainingProvider(
```

Comment:
Not sure the previous URL would actually work because of the commitmentStatementId (might be a binding issue) and how would it know which commitment statement to update.

As for the new URL. If we want to use a rest style the commitmentstatementId should really be prefixed with  /commitmentstatement/{commitmentstatementId}/...

Would it be better if the commitmentstatementId was in the body of the request. It is in essence a value denoting the latest commitment statement, the person has just viewed. And is therefore an attribute of teh apprenticeship 

Another thought, if we want to update the commitmentstatement explicitly, without caring about whether this is the latest CS. we could use the following

 [HttpPost("commitmentstatements/{commitmentstatementId}")
 
 this would achieve the same as the existing code and would be much simplier.  The only -ve is you are relying on our FE code to be sending the correct id for this apprenticeship

Reply:
As discussed this morning, I see the commitment statement ID as the identifier of the commitment statement, and as such belongs to the route.  Leaving the apprenticeship and apprenticeship IDs in the route prevent errors updating a statement for completely the wrong person / apprenticeship, as you say.

Reply:
I have added "statements" segment to the routes, so they are now

    /apprentices/{id}/apprenticeships/{id}/statements/{id}/trainingproviderconfirmation